packages/@react-spectrum/s2/src/RangeCalendar.tsx

```diff
@@ -10,38 +10,390 @@
+  },
+  contain: 'inline-size',
+  paddingTop: '--field-gap',
+  cursor: {
+    default: 'text',
+    isDisabled: 'default'
+  }
+});
+
+export const RangeCalendar = /*#__PURE__*/ (forwardRef as forwardRefType)(function RangeCalendar<T extends DateValue>(props: RangeCalendarProps<T>, ref: ForwardedRef<HTMLDivElement>) {
+  [props, ref] = useSpectrumContextProps(props, ref, RangeCalendarContext);
+  let {
+    visibleMonths = 1,
+    errorMessage,
+    UNSAFE_style,
+    UNSAFE_className,
+    styles,
+    ...otherProps
+  } = props;
+
   return (
-    <AriaRangeCalendar {...props}>
-      <header>
-        <Button slot="previous">◀</Button>
-        <Heading />
-        <Button slot="next">▶</Button>
-      </header>
-      <CalendarGrid>
-        {(date) => <CalendarCell date={date} />}
-      </CalendarGrid>
-      {errorMessage && <Text slot="errorMessage">{errorMessage}</Text>}
+    <AriaRangeCalendar
+      {...otherProps}
+      ref={ref}
+      visibleDuration={{months: visibleMonths}}
+      style={UNSAFE_style}
+      className={(UNSAFE_className || '') + calendarStyles(null, styles)}>
+      {({isInvalid, isDisabled}) => {
+        return (
+          <>
+            <Header styles={headerStyles}>
+              <CalendarButton slot="previous"><ChevronLeftIcon /></CalendarButton>
+              <CalendarHeading />
+              <CalendarButton slot="next"><ChevronRightIcon /></CalendarButton>
+            </Header>
+            <div
+              className={style({
+                display: 'flex',
+                flexDirection: 'row',
+                gap: 24,
+                width: 'full'
+              })}>
+              {Array.from({length: visibleMonths}).map((_, i) => {
+                return (
+                  <CalendarGrid offset={{months: i}} key={i} className={style({borderSpacing: 0, borderCollapse: 'collapse'})}>
+                    <CalendarGridHeader>
+                      {(day) => (
+                        <CalendarHeaderCell className={headerCellStyles}>
+                          {day}
+                        </CalendarHeaderCell>
+                      )}
+                    </CalendarGridHeader>
+                    <CalendarGridBody>
+                      {(date, weekIndex, dayIndex) => {
+                        return (
+                          <CalendarCell date={date} weekIndex={weekIndex} dayIndex={dayIndex} />
+                        );
+                      }}
+                    </CalendarGridBody>
+                  </CalendarGrid>
+                );
+              })}
+            </div>
+            {errorMessage && (
+              <Text slot="errorMessage" className={helpTextStyles({isInvalid, isDisabled})}>
+                {/* @ts-ignore */}
+                {errorMessage}
+              </Text>
+            )}
+          </>
+        );
+      }}
     </AriaRangeCalendar>
   );
-}
+});
+
+// Ordinarily the heading is a formatted date range, ie January 2025 - February 2025.
```

Comment:
Lot's more duplication. This shouldn't be necessary right? There are no differences with normal Calendars